backend/ionic.py

```diff
@@ -0,0 +1,71 @@
+import json
+import logging
+import requests
+import uuid
+
+
+IONIC_URL = 'https://api.ionic.io'
+
+
+def BuildUrl(suffix):
+  return '%s/%s' % (IONIC_URL, suffix)
+
+
+class Ionic(object):
+  """Ionic API Wrapper."""
+
+  def __init__(self, app_id, security_tag, access_token):
+    self.app_id = app_id
+    self.access_token = 'Bearer %s' % access_token
+    self.security_tag = security_tag
+    self.Test()
+
+  def GetAuthHeaders(self):
+    """Generate authoriztaion headers."""
+    return {
+      'Authorization': self.access_token,
+      'Content-Type': 'application/json',
+    }
+
+  def Test(self):
+    """Validate credentials.
+
+    Returns:
+      None. Raises exception on error.
+    """
+    r = requests.get(BuildUrl('auth/test'), headers=self.GetAuthHeaders())
+    r.raise_for_status()
+
+  def Post(self, url_suffix, json_data):
+    """Post data to ionic."""
+    r = requests.post(BuildUrl(url_suffix), json=json_data, headers=self.GetAuthHeaders())
+    r.raise_for_status()
+    return r
+
+  def SendNotification(self, tokens, message, destination):
+    """Send a notification to the device tokens specified.
+
+    Args:
+      tokens: An array of device tokens to send to.
+      message: Message to send in the notification.
+      destination: Destination the notificatoin should open.
+
+    Returns:
+      Notification response. Raises HTTPError on exception.
+    """
+    data = {
+      'app_id': self.app_id,
+      'notification': {
+        'message': message,
+        'payload': {
+          'destination': destination,
+        }
+      },
+      'profile': self.security_tag
+    }
+    if tokens:
+      data['tokens'] = tokens
+    else:
+      data['send_to_all'] = True
```

Comment:
Do we ever want to do this? If not, might be better to not have it at all so when we accidentally set tokens=None we don't spam people.